stats/stats/src/data_source/kinds/local_db/mod.rs

```diff
@@ -260,6 +264,32 @@ where
         Ok(())
     }
 
+    async fn set_next_update_from_itself(
```

Comment:
actually need to test this with counters (see if it breaks them, it's possible actually)

Reply:
it actually breaks them. the error is the same as was in `https://github.com/blockscout/blockscout-rs/issues/1125` but now the "incorrect" state is introduced on purpose instead of accidentally. I probably know how to fix it, let me look into this.